src/Microsoft.DocAsCode.Build.Engine/PostProcessors/SitemapGenerator.cs

```diff
@@ -0,0 +1,167 @@
+
+            if (manifest.SitemapOptions.Priority.HasValue && (manifest.SitemapOptions.Priority < 0 || manifest.SitemapOptions.Priority > 1))
+            {
+                Logger.LogWarning($"Invalid priority {manifest.SitemapOptions.Priority}, priority must be between 0.0 and 1.0. Use default value 0.5 instead");
+                manifest.SitemapOptions.Priority = 0.5;
+            }
+
+            XElement urlset = new XElement(XName.Get("urlset", NamespaceName));
+            XDocument sitemapDocument = new XDocument(urlset);
+
+            foreach (var file in GetHtmlOutputFiles(manifest).OrderBy(s => s.Item1))
+            {
+                var options = GetOptions(manifest.SitemapOptions, file.Item1);
+
+                var currentBaseUri = baseUri;
+                if (options.BaseUrl != manifest.SitemapOptions.BaseUrl)
+                {
+                    if (!Uri.TryCreate(options.BaseUrl, UriKind.Absolute, out currentBaseUri))
+                    {
+                        Logger.LogWarning($"Base url {options.BaseUrl} is not in a valid uri format, use base url from the default setting {manifest.SitemapOptions.BaseUrl} instead.");
+                        currentBaseUri = baseUri;
+                    }
+                }
+
+                urlset.Add(GetElement(file.Item2.RelativePath, currentBaseUri, options));
+            }
+
+            var sitemapOutputFile = Path.Combine(outputFolder, SitemapName);
+            Logger.LogInfo($"Sitemap file is successfully exported to {sitemapOutputFile}");
+            sitemapDocument.Save(sitemapOutputFile);
+            return manifest;
+        }
+
+        private IEnumerable<Tuple<string, OutputFileInfo>> GetHtmlOutputFiles(Manifest manifest)
+        {
+            if (manifest.Files == null)
+            {
+                yield break;
+            }
+
+            foreach(var file in manifest.Files)
+            {
+                if (file.DocumentType != "Toc"
+                    && file.OutputFiles.TryGetValue(HtmlExtension, out var info) 
+                    && !string.IsNullOrEmpty(info.RelativePath))
+                {
+                    yield return Tuple.Create(file.SourceRelativePath, info);
+                }
+            }
+        }
+
+        private XElement GetElement(string relativePath, Uri baseUri, SitemapElementOptions options)
+        {
+            var uri = new Uri(baseUri, relativePath);
+
+            return new XElement
+                 (XName.Get("url", NamespaceName),
+                 new XElement(XName.Get("loc", NamespaceName), uri.AbsoluteUri),
+                 new XElement(XName.Get("lastmod", NamespaceName), (options.LastModified ?? DateTime.Now).ToString("yyyy-MM-ddThh:mm:ssK")),
+                 new XElement(XName.Get("changefreq", NamespaceName), (options.ChangeFrequency ?? PageChangeFrequency.Daily).ToString().ToLowerInvariant()),
+                 new XElement(XName.Get("priority", NamespaceName), options.Priority ?? 0.5)
```

Comment:
changefreq and priority is optional, remove default value?

Reply:
prefer to keep our own default value